core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -383,60 +393,179 @@ object Battle {
+            }
         }
-
-        for (civ in attacker.getCivInfo().getKnownCivs()) {
-            civ.getDiplomacyManager(attackingCiv)
-                    .setModifier(DiplomaticModifiers.UsedNuclearWeapons, -50f)
+    }
+    
+    private fun nukeStrength2Effect(attacker: MapUnitCombatant, tile: TileInfo) {
+        // https://forums.civfanatics.com/threads/unit-guide-modern-future-units-g-k.429987/#2
+        // https://www.carlsguides.com/strategy/civilization5/units/aircraft-nukes.php
+        // Testing done by Ravignir
+        var damageModifierFromMissingResource = 1f
+        val civResources = attacker.getCivInfo().getCivResourcesByName()
+        for (resource in attacker.unit.baseUnit.getResourceRequirements().keys) {
+            if (civResources[resource]!! < 0 && !attacker.getCivInfo().isBarbarian())
+                damageModifierFromMissingResource *= 0.5f // I could not find a source for this number, but this felt about right
         }
-
-        // Instead of postBattleAction() just destroy the missile, all other functions are not relevant
-        if ((attacker as MapUnitCombatant).unit.hasUnique("Self-destructs when attacking")) {
-            attacker.unit.destroy()
+        // Damage and/or destroy cities
+        val city = tile.getCity()
+        if (city != null && city.location == tile.position) {
+            var populationLoss = 1f;
+            for (unique in city.civInfo.getMatchingUniques("Population loss from nuclear attacks -[]%")) {
+                populationLoss *= 1 - unique.params[0].toFloat() / 100f
+            }
+            if (city.population.population < 5) {
+                city.destroyCity()
+            } else {
+                city.population.population = (city.population.population * (0.6 + Random().nextFloat() * 0.2) * populationLoss).toInt()
+                if (city.population.population < 5 && populationLoss == 1f) city.population.population = 5
+                if (city.population.population < 1) city.population.population = 1
+                city.population.unassignExtraPopulation()
+                city.health -= (0.5 * city.getMaxHealth() * damageModifierFromMissingResource).toInt()
+                if (city.health < 1) city.health = 1
+            }
+            postBattleNotifications(attacker, CityCombatant(city), city.getCenterTile())
+            destroyIfDefeated(city.civInfo, attacker.getCivInfo())
+        }
+        // Destroy all units
+        for (unit in tile.getUnits()) {
+            unit.destroy()
+            postBattleNotifications(attacker, MapUnitCombatant(unit), unit.currentTile)
+            destroyIfDefeated(unit.civInfo, attacker.getCivInfo())
+        }
+        // Remove improvements
+        tile.improvement = null
+        tile.improvementInProgress = null
+        tile.turnsToImprovement = 0
+        tile.roadStatus = RoadStatus.None
+        if (tile.isLand && !tile.isImpassible() && !tile.terrainFeatures.contains("Fallout") &&
```

Comment:
It looks like unlike strength 1, we don't remove the features?

Reply:
My bad, must have missed this, I'll fix it later today